enhancements/baremetal/baremetal-provisioning-optional.md

```diff
@@ -0,0 +1,179 @@
+| dnsmasq               | X             | TFTP Only     |                         |
+| httpd                 | X             | X             | X                       |
+| ironic-api            | X             | X             | X                       |
+| ironic-conductor      | X             | X             | X                       |
+| ironic-inspector      | X             | X             | X                       |
+| machine-os-downloader | X             | X             | X                       |
+| static-ip-manager     | X             | X             | X                       |
+
+### User Stories
+
+- As a user, I want to be able to disable the provisioning network using
+  an installer provisioningNetwork field.
+
+- As a user, I want to be able to configure external DHCP using the new
+  profile or the older boolean flag, with the older flag being deprecated.
+  See https://github.com/openshift/installer/pull/2829 for a deprecation
+  example.
+
+- As a user, I want the provisioningNetwork field from the installer to
+  be propogated to the cluster Provisioning custom resource.
+
+### Implementation Details/Notes/Constraints
+
+Today, we rely on two IP's for bootstrap and the cluster to host the
+Ironic services. Without the dedicated provisioning network, these will
+need to be static IP's on the external network.
+
+We may be able to use the API VIP, however, both provisioning
+infrastructures in the bootstrap and cluster can (and often are) online
+at the same time. That is to say, the third control plane member is
+still provisioning while the machine-api-operator is already busy
+provisioning worker nodes. To remove the two IP's, we would need to
+constrain this and not bring up the provisioning infrastructure until
+the third control plane member is online.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+Upgrades will need to handle upgrading the Provisioning CR from the
+previous version to the latest.
+
+### Version Skew Strategy
+
+MAO/CBO will need to understand both versions of the Provisioning CR,
+and be able to look at the older provisioningDHCPExternal field.
+
+## Implementation History
+
+
+## Drawbacks
```

Comment:
Maybe discuss the security impact of moving everything on to the external network?

One example -- if this requires moving the Ironic API endpoint onto the external network, I imagine it needs to be secured?  We cheated and left the API wide open since it was on the isolated provisioning network.  (the introspection API anyway, I know the normal API is hidden behind baremetal-operator)

Reply:
Added a blurb about this